src/Microsoft.Health.Dicom.Blob/Extensions/AzureStorageErrorExtensions.cs

```diff
@@ -0,0 +1,51 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Azure;
+using Azure.Storage.Blobs.Models;
+
+namespace Microsoft.Health.Dicom.Blob.Extensions;
+
+internal static class AzureStorageErrorExtensions
+{
+    private static readonly List<BlobErrorCode> Customer403ErrorCodes = new List<BlobErrorCode>
+    {
+        BlobErrorCode.AuthorizationFailure,
+        BlobErrorCode.AuthorizationPermissionMismatch,
+        BlobErrorCode.InsufficientAccountPermissions,
+        BlobErrorCode.AccountIsDisabled,
+        "KeyVaultEncryptionKeyNotFound",
+        "KeyVaultAccessTokenCannotBeAcquired",
+        "KeyVaultVaultNotFound",
+    };
+
+    private static readonly List<BlobErrorCode> Customer404ErrorCodes = new List<BlobErrorCode>
+    {
+        BlobErrorCode.ContainerNotFound,
+        "FilesystemNotFound",
+    };
+
+    private static readonly List<BlobErrorCode> Customer409ErrorCodes = new List<BlobErrorCode>
+    {
+        BlobErrorCode.ContainerBeingDeleted,
+        BlobErrorCode.ContainerDisabled,
+    };
+
+    public static bool IsConnectedStoreCustomerError(this RequestFailedException rfe)
+    {
+        return (rfe.Status == 403 && Customer403ErrorCodes.Any(e => e.ToString().Equals(rfe.ErrorCode, StringComparison.OrdinalIgnoreCase))) ||
```

Comment:
The reason I called ToString() here is so we could use the `StringComparison.OrdinalIgnoreCase`. I'm worried since some error codes we are still referencing by string, if the case changes then we won't match on it.